common/src/main/java/com/microsoft/identity/common/internal/telemetry/events/ApiStartEvent.java

```diff
@@ -0,0 +1,204 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.telemetry.events;
+
+import android.support.annotation.NonNull;
+
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.request.AcquireTokenOperationParameters;
+import com.microsoft.identity.common.internal.request.AcquireTokenSilentOperationParameters;
+import com.microsoft.identity.common.internal.request.BrokerAcquireTokenOperationParameters;
+import com.microsoft.identity.common.internal.request.BrokerAcquireTokenSilentOperationParameters;
+import com.microsoft.identity.common.internal.request.OperationParameters;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.NoSuchAlgorithmException;
+
+import static com.microsoft.identity.common.internal.telemetry.TelemetryEventStrings.*;
+
+public class ApiStartEvent extends BaseEvent {
+    private static final String TAG = ApiStartEvent.class.getSimpleName();
+
+    public ApiStartEvent() {
+        super();
+        names(TELEMETRY_EVENT_API_EVENT_START);
+    }
+
+    public ApiStartEvent putProperties(OperationParameters parameters) {
+        this.put(TELEMETRY_KEY_AUTHORITY, parameters.getAuthority().toString());
+        this.put(TELEMETRY_KEY_AUTHORITY_TYPE, parameters.getAuthority().getAuthorityTypeString());
+        this.put(TELEMETRY_KEY_APPLICATION_NAME, parameters.getApplicationName());
+        this.put(TELEMETRY_KEY_APPLICATION_VERSION, parameters.getApplicationVersion());
+        this.put(TELEMETRY_KEY_SDK_NAME, parameters.getSdkType().name());
+        this.put(TELEMETRY_KEY_SDK_VERSION, parameters.getSdkVersion());
+        this.put(TELEMETRY_KEY_CLAIM_REQUEST, StringUtil.isEmpty(
+                parameters.getClaimsRequestJson())? TELEMETRY_VALUE_NO : TELEMETRY_VALUE_YES
+        );
+        this.put(TELEMETRY_KEY_REDIRECT_URI, parameters.getRedirectUri());
+        this.put(TELEMETRY_KEY_CLIENT_ID, parameters.getClientId());
+        this.put(TELEMETRY_KEY_SCOPE_SIZE, String.valueOf(parameters.getScopes().size()));
+        this.put(TELEMETRY_KEY_SCOPE, parameters.getScopes().toString());
```

Comment:
Do these properties align with what iOS is planning to emit?

Reply:
It contains what iOS already has and I also added other info into the Api events based our previous icm experience, including
TELEMETRY_KEY_AUTHORITY
TELEMETRY_KEY_AUTHORITY_TYPE
TELEMETRY_KEY_CLAIM_REQUEST
TELEMETRY_KEY_REDIRECT_URI
TELEMETRY_KEY_SCOPE_SIZE
TELEMETRY_KEY_SCOPE
TELEMETRY_KEY_REQUEST_QUERY_PARAMS
TELEMETRY_KEY_IS_FORCE_REFRESH
TELEMETRY_KEY_BROKER_PROTOCOL_VERSION

Please let me know which you want to remove or add.